src/Utilities/DoubleDict.jl

```diff
@@ -0,0 +1,314 @@
+module DoubleDicts
+
+import MathOptInterface
+
+const MOI = MathOptInterface
+const CI{F,S} = MOI.ConstraintIndex{F,S}
+
+"""
+    DoubleDict
+
+Special dictionary to map ConstraintIndex to ConstraintIndex with minimal
+performance loss from the type instability due different constraint types.
+"""
+mutable struct DoubleDict{D, D2}
```

Comment:
It seems weird to do `D` here and `D{Int, Int}` in the field, I would expect something like
`D<:AbstractDict{Int, Int}, D2<:AbstractDict{Tuple{DataType, DataType}, D}` and a field `dict::D2`

Reply:
yup, that was messy, thanks for the cleaup!